### ISSUE TITLE
Incremental update misses changes in symlinks

### ISSUE THREAD

**Issue by joliss** [broccolijs/broccoli-merge-trees#61]
Hi @stefanpenner, @hjdivad, @rwjblue,

I encountered the following bug: If there is a symlink in one of the input directories, say

```
input-dir-1/symlink -> /.../foo-dir
```

and it changes to point to a different directory, e.g.

```
input-dir-1/symlink -> /.../bar-dir
```

the output symlink isn't updated, i.e. it continues to be

```
output-dir/symlink -> /.../foo-dir
```

Between the logic in [node-merge-trees](https://github.com/broccolijs/node-merge-trees) and [fs-tree-diff](https://github.com/stefanpenner/fs-tree-diff) I'm pretty confused about where the problem is occurring, so I'm not submitting a test case here, but perhaps one of you lovely folks knows how to isolate (and fix) this problem?

**Comment by stefanpenner** [comment]
> so I'm not submitting a test case here

If you have time to, that would likely make it super easy for me to debug. That way, I know I am hitting exactly what you are while debugging.

**Comment by hjdivad** [comment]
@joliss there's a couple things that might make things a bit easier.  Obviously a failing test would be ideal, but otherwise

1. `npm ls` output (for versions)
2. Log the patch operation and relativePath inside the loop of [_applyPatch](https://github.com/broccolijs/node-merge-trees/blob/master/index.js#L85) to help isolate the problem to either fs-tree-diff or node-merge-trees

**Comment by joliss** [comment]
This is fixed in broccoli-merge-trees 3.0.0 / node-merge-trees 2.0.0.

The tests for this are covered in [node-fs-updater](https://github.com/broccolijs/node-fs-updater)'s test suite, since that's where all the incremental-updating code now lives.
